diff --git a/Modules/Test/classes/class.ilTestOutputGUI.php b/Modules/Test/classes/class.ilTestOutputGUI.php
--- a/Modules/Test/classes/class.ilTestOutputGUI.php	
+++ b/Modules/Test/classes/class.ilTestOutputGUI.php	(date 1664186882294)
@@ -231,6 +231,24 @@
         $this->ctrl->setParameter($this, 'sequence', $sequenceElement);
         $this->ctrl->setParameter($this, 'pmode', '');
 
+        /**
+         * @author Internetlehrer-GmbH
+         * @package Events2Lrs
+         */
+        global $DIC;  /** @var \ILIAS\DI\Container $DIC */
+
+        $DIC->event()->raise(
+            'Plugins/Event2Lrs',
+            'startTestPass',
+            [
+                'active_id' => $this->testSession->getActiveId(),
+                'pass' => $this->testSession->getPass(),
+                'user_id' => $this->testSession->getUserId(),
+                'ref_id' => $this->object->getRefId()
+            ]
+        );
+        /** @package EOF Events2Lrs */
+
         if ($this->object->getListOfQuestionsStart()) {
             $this->ctrl->redirect($this, ilTestPlayerCommands::QUESTION_SUMMARY);
         }
@@ -829,6 +847,24 @@
             $this->testSequence->saveToDb();
         }
 
+        /**
+         * @author Internetlehrer-GmbH
+         * @package Events2Lrs
+         */
+        global $DIC;  /** @var \ILIAS\DI\Container $DIC */
+
+        $DIC->event()->raise(
+            'Plugins/Events2Lrs',
+            'resumeTestPass',
+            [
+                'active_id' => $this->testSession->getActiveId(),
+                'pass' => $this->testSession->getPass(),
+                'user_id' => $this->testSession->getUserId(),
+                'ref_id' => $this->object->getRefId()
+            ]
+        );
+        /** @package EOF Events2Lrs */
+
         if ($this->object->getListOfQuestionsStart()) {
             $this->ctrl->redirect($this, ilTestPlayerCommands::QUESTION_SUMMARY);
         }
